subprojects/signing/src/main/java/org/gradle/plugins/signing/signatory/pgp/PgpSignatoryFactory.java

```diff
@@ -81,8 +81,16 @@ protected PgpSignatory readProperties(Project project, String prefix, String nam
         for (String property : PROPERTIES) {
             String qualifiedProperty = (String)getQualifiedPropertyName(prefix, property);
             if (project.hasProperty(qualifiedProperty)) {
-                values.add(
-                    project.property(qualifiedProperty));
+                Object prop = project.property(qualifiedProperty);
+                if(prop != null){
+                    values.add(prop);
+                } else {
+                    if(required){
```

Comment:
This logic is duplicated and should be reused. The `else` block below it uses the same logic.